[CON-423] feat: (zoho)update write input dynamically

Jkarage opened the pull request:
### Issue
-  Zoho expects a list of objects that's are to be created/updated.  
- Ampersand API sends a map[string]any. A single object

This PR wraps the object into a slice and sends it. Checks if the received data is not a list.

Tests:
<img width="1219" alt="Screenshot 2024-12-19 at 12 03 16" src="https://github.com/user-attachments/assets/2c5ae868-def1-41bc-a33a-f2b55f19552d" />

For more context: https://linear.app/ampersand/issue/CON-420/zoho-crm-deep-connector

jlimatampersand commented on providers/zohocrm/write.go:
```diff
@@ -67,10 +69,13 @@ func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*comm
+	// If not, we wrap the data in a slice, else we send it as is.
+	recordDataCopy, ok := config.RecordData.([]map[string]any)
+	if !ok {
+		body = map[string]any{"data": []any{config.RecordData}}
```
I would also assert `config.RecordData.(map[string]any)` here and error out if not ok.

jlimatampersand reviewed (approved):
Preapproving to unblock, but make sure implement type assertion and return error if needed.

Jkarage commented:
@jlimatampersand Addressed your review, and returned `common.ErrBadRequest` if the assertion of the map fails. Not so comfortable with the error type returned.  Merging so I can continue with Integration testing. Will update this if you feel the need to.

jlimatampersand commented on providers/zohocrm/write.go:
```diff
@@ -83,18 +84,30 @@ func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*comm
+	if !ok {
+		objectData, ok := payload.(map[string]any)
+		if !ok {
+			return nil, common.ErrBadRequest
```
@Jkarage Can you make an update PR that wraps this error?

```
return nil, fmt.Errorf("%w: unexpected input type. Expecting either a json object or a list of objects, but received type '%T'"), payload) 
```
